wordpress-mobile/WordPress-Android
# Plans in site creation: Remove domain purchasing feature flag

Fixes #19230 

This PR removes domain purchasing in site creation feature flag, and replaces it with plans in site creation feature flag.


To test:

Test 1

- `Launch` JP app
- `Go` to Me-> Debug settings
- `Switch` to `Features in Development` tab
- `Verify` there's no `SiteCreationDomainPurchasing` feature flag is listed there
- `Switch` to `Remote Feature` tab, find `plans_in_site_creation` and enable it

Test 2

- `Go` to **My Site** tab
- `Tap` the down arrow to the right of the site name on the My Site screen
- `Tap` the + icon at the top right on the Site picker screen
- `Select` ⊕ **Create WordPress.com site** on the modal dialog
- `Tap` Skip at top right on the first step (topic screen)
- `Tap` Skip at top right on the next step (theme screen)
- `Enter` a search query (whatever you want here)
- `Verify` that result domains have either `Free` or `Price are no prices or "free" listed next to the domains on the right side

>**Note**
> Selecting a custom domain (with price) will launch, site assembly followed by checkout
> Selecting a free domain will launch site assembly an done

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual and unit tests

3. What automated tests I added (or what prevented me from doing so)
Update existing unit tests

PR submission checklist:
